cpp/benchmarks/string/experimental/stringview_compare.cu

```diff
@@ -0,0 +1,503 @@
+  }
+};
+
+/**
+ * Compares two strings from ArrowBinaryView objects.
+ */
+struct compare_arrow_sv {
+  ArrowBinaryView* d_items;
+  char const* d_chars;
+  __device__ bool operator()(cudf::size_type lhs, cudf::size_type rhs)
+  {
+    auto& item_lhs = d_items[lhs];
+    auto& item_rhs = d_items[rhs];
+
+    // shortcut to check preview bytes
+    auto pv_lhs = reinterpret_cast<uint32_t const*>(item_lhs.inlined.data)[0];
+    auto pv_rhs = reinterpret_cast<uint32_t const*>(item_rhs.inlined.data)[0];
+    if (pv_lhs != pv_rhs) {
+      return cudf::hashing::detail::swap_endian(pv_lhs) <
+             cudf::hashing::detail::swap_endian(pv_rhs);
+    }
+
+    // prefix matches so check how many bytes are left to compare
+    constexpr auto prefix_size = static_cast<cudf::size_type>(sizeof(uint32_t));
+    auto const size_lhs        = item_lhs.inlined.size;
+    auto const size_rhs        = item_rhs.inlined.size;
+    // if no bytes left to compare, we are done (strings are equal)
+    if (size_lhs <= prefix_size && size_rhs <= prefix_size) { return false; }
+
+    // compare the remaining bytes
+    auto const d_str_lhs = cudf::string_view(
+      get_string_view(item_lhs, d_chars).data() + prefix_size, size_lhs - prefix_size);
+    auto const d_str_rhs = cudf::string_view(
+      get_string_view(item_rhs, d_chars).data() + prefix_size, size_rhs - prefix_size);
+
+    return d_str_lhs < d_str_rhs;
+  }
+};
+
+#else
+
+/**
+ * Hashes a string from a cudf column
+ */
+struct hash_sv {
+  cudf::column_device_view d_strings;
+  __device__ cudf::hash_value_type operator()(cudf::size_type idx) const
+  {
+    auto const d_str  = d_strings.element<cudf::string_view>(idx);
+    auto const hasher = cudf::hashing::detail::MurmurHash3_x86_32<cudf::string_view>{0};
+    return hasher(d_str);
+  }
+};
+
+/**
+ * Checks if a string from a cudf column starts with a target string
+ */
+struct starts_sv {
+  cudf::column_device_view d_strings;
+  cudf::size_type tgt_size;
+  __device__ bool operator()(cudf::size_type idx) const
+  {
+    auto const d_str = d_strings.element<cudf::string_view>(idx);
+    auto const d_tgt = cudf::string_view("abcdefghijklmnopqrstuvwxyz", tgt_size);
+    return d_str.size_bytes() >= d_tgt.size_bytes() &&
+           d_tgt.compare(d_str.data(), d_tgt.size_bytes()) == 0;
+  }
+};
+
+/**
+ * Compares two strings from a cudf column
+ */
+struct compare_sv {
+  cudf::column_device_view d_strings;
+  __device__ bool operator()(cudf::size_type lhs, cudf::size_type rhs)
+  {
+    auto const d_str_lhs = d_strings.element<cudf::string_view>(lhs);
+    auto const d_str_rhs = d_strings.element<cudf::string_view>(rhs);
+    return d_str_lhs < d_str_rhs;
+  }
+};
+
+#endif
+
+}  // namespace
+
+#if BM_ARROW_STRINGVIEW
+
+/**
+ * Creates an ArrowBinaryView vector and data buffer from a strings column.
+ */
+static std::pair<rmm::device_uvector<ArrowBinaryView>, rmm::device_buffer> create_sv_array(
```

Comment:
can this also be in the anon namespace? If so, I think we can further merge the conditional directives